### PR TITLE
Support of large integers

### PR DESCRIPTION
**Pull request by rafaelsamenezes**
This PR adds support for 128 bits integers.

### CODE REVIEW

**Review by lucasccordeiro — APPROVED** [review]
LGTM.

File: src/util/c_typecast.cpp [hunk 1]
```diff
@@ -361,7 +361,9 @@ c_typecastt::c_typet c_typecastt::get_c_type(const type2tc &type)
     signedbv_type2tc signed_type = type;
     unsigned width = signed_type->width;
 
-    if(width <= config.ansi_c.char_width)
+    if(width == 128)
```
Comment (mikhailramalho) [inline src/util/c_typecast.cpp]: Can we just remove the magic number, i.e., define them in the config object?

**Review by mikhailramalho — CHANGES_REQUESTED** [review]

File: src/util/c_types.h [hunk 2]
```diff
@@ -17,10 +17,16 @@ type2tc index_type2();
...
 typet int_type();
 type2tc int_type2();
-
 typet uint_type();
 type2tc uint_type2();
-
+typet int128_type();
+type2tc int128_type2();
+typet uint_type();
+type2tc uint_type2();
```
Comment (fbrausse) [inline src/util/c_types.h]: `uint_type[2]()` have already been declared above.

File: src/util/c_types.h [hunk 3]
```diff
@@ -17,10 +17,16 @@ type2tc index_type2();
...
+typet uint128_type();
+type2tc uint128_type2();
+typet signed_short_int_type();
+typet unsigned_short_int_type();
```
Comment (fbrausse) [inline src/util/c_types.h]: Same duplication of declaration.

File: src/util/c_types.h (hunk 2 above)
Comment (rafaelsamenezes) [reply src/util/c_types.h]: ow, nice catch!

**Review by mikhailramalho — COMMENTED** [review]
LGTM!

**Review by mikhailramalho — APPROVED** [review]

File: src/util/c_types.cpp [hunk 4]
```diff
@@ -142,6 +142,26 @@ type2tc long_long_int_type2()
   return get_int_type(config.ansi_c.long_long_int_width);
 }
 
+typet int128_type()
+{
+  return signedbv_typet(128);
```
Comment (fbrausse) [inline src/util/c_types.cpp:147]: Sorry, have missed this. Should we use the new config setting `int_128_width` here and in the functions below?

**Pull request merged**